File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Members/Properties.vb
#Region "Class"

Namespace Class
    Public Class TypeWithNoProperties
    End Class

    Public MustInherit Class AbstractTypeWithProperties
        Public Overridable Property SealedProperty() As Int32
    End Class

    Public Class TypeWithProperties
        Inherits AbstractFooType

        Public Property AutoProperty() As Int32

        Public ReadOnly Property ReadOnlyProperty() As Int32

        Public WriteOnly Property WriteOnlyProperty() As Int32
            Set
            End Set
        End Property

        Public Property WithGetterModifierProperty() As Int32
            Friend Get
                Return 1
            End Get
            Set(value As Int32)
            End Set
        End Property

        Public Property WithSetterModifierProperty() As Int32
            Get
                Return 1
            End Get
            Friend Set(value As Int32)
            End Set
        End Property

        Public ReadOnly Property WithInitialValueProperty() As Int32 = 1

        Private Property PrivateProperty() As Int32

        Protected Property ProtectedProperty() As Int32

        Friend Property InternalProperty() As Int32

        Protected Friend Property ProtectedInternalProperty() As Int32

        Property WithNoModifierProperty() As Int32

        Public Shared Property StaticProperty() As Int32

        Public Overridable Property VirtualProperty() As Int32

        Public NotOverridable Overrides Property SealedProperty() As Int32
    End Class
End Namespace

#End Region

#Region "Struct"

Namespace Struct
    Public Structure TypeWithNoProperties
    End Structure

    Public Structure TypeWithProperties
        Public Property AutoProperty() As Int32

        Public ReadOnly Property ReadOnlyProperty() As Int32

        Public WriteOnly Property WriteOnlyProperty() As Int32
            Set
            End Set
        End Property

        Public Property WithGetterModifierProperty() As Int32
            Friend Get
                Return 1
            End Get
            Set(value As Int32)
            End Set
        End Property

        Public Property WithSetterModifierProperty() As Int32
            Get
                Return 1
            End Get
            Friend Set(value As Int32)
            End Set
        End Property

        Public Shared ReadOnly Property WithInitialValueProperty() As Int32 = 1

        Private Property PrivateProperty() As Int32

        Friend Property InternalProperty() As Int32

        Property WithNoModifierProperty() As Int32

        Public Shared Property StaticProperty() As Int32
    End Structure
End Namespace

#End Region

#Region "Interface"

Namespace Interface
    Public Interface TypeWithNoProperties
    End Interface

    Public Interface TypeWithProperties
        Property AutoProperty() As Int32

        ReadOnly Property ReadOnlyProperty() As Int32

        WriteOnly Property WriteOnlyProperty() As Int32

        Property WithNoModifierProperty()
    End Interface
End Namespace

#End Region

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Members/Properties.cs
#region Class

namespace Class
{
    public class TypeWithNoProperties
    {
    }

    public class TypeWithPrivateDefaultModifier
    {
        Int32 WithNoModifierProperty { get; set; }
    }

    public class TypeWithProperties
    {
        public Int32 AutoProperty { get; set; }

        public Int32 ReadOnlyProperty { get; }

        public Int32 WriteOnlyProperty
        {
            set { }
        }

        public Int32 WithExpressionBody => 1;

        public Int32 WithGetterModifierProperty
        {
            internal get { return 1; }
            set { }
        }

        public Int32 WithSetterModifierProperty
        {
            get { return 1; }
            internal set { }
        }

        public Int32 WithInitialValueProperty { get; } = 1;

        private Int32 PrivateProperty { get; set; }

        internal Int32 InternalProperty { get; set; }

        public static Int32 StaticProperty { get; set; }

        protected Int32 ProtectedProperty { get; set; }

        protected internal Int32 ProtectedInternalProperty { get; set; }

        public virtual Int32 VirtualProperty { get; set; }

        public sealed override Int32 SealedProperty { get; set; }
    }
}

#endregion

#region Interface

namespace Interface
{
    public interface TypeWithNoProperties
    {
    }

    public interface TypeWithPublicDefaultModifier
    {
        Int32 WithNoModifierProperty { get; set; }
    }

    public interface TypeWithProperties
    {
        Int32 AutoProperty { get; set; }

        Int32 ReadOnlyProperty { get; }

        Int32 WriteOnlyProperty { set; }
    }
}
#endregion

#region Struct

namespace Struct
{
    public struct TypeWithNoProperties
    {
    }

    public struct TypeWithPrivateDefaultModifier
    {
        Int32 WithNoModifierProperty { get; set; }
    }

    public struct TypeWithProperties
    {
        public Int32 AutoProperty { get; set; }

        public Int32 ReadOnlyProperty { get; }

        public Int32 WriteOnlyProperty
        {
            set { }
        }

        public Int32 WithExpressionBody => 1;

        public Int32 WithGetterModifierProperty
        {
            internal get { return 1; }
            set { }
        }

        public Int32 WithSetterModifierProperty
        {
            get { return 1; }
            internal set { }
        }

        public static Int32 WithInitialValueProperty { get; } = 1;

        private Int32 PrivateProperty { get; set; }

        internal Int32 InternalProperty { get; set; }

        public static Int32 StaticProperty { get; set; }
    }
}

#endregion